extensions/amp-jwplayer/0.1/amp-jwplayer.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {setStyles} from '../../../src/style';
+import {user} from '../../../src/log';
+
+class AmpJWPlayer extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    // Host that serves player configuration and content redirects
+    this.preconnect.url('https://content.jwplatform.com', onLayout);
+    // CDN which hosts jwplayer assets
+    this.preconnect.url('https://ssl.p.jwpcdn.com', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+
+    /** @private @const {number} */
+    this.width_ = getLengthNumeral(width);
+
+    /** @private @const {number} */
+    this.height_ = getLengthNumeral(height);
+
+    if (!this.getPlaceholder()) {
+      this.buildImagePlaceholder_();
+    }
+  }
+
+
+  /** @override */
+  layoutCallback() {
+    /** @private @const {string} */
```

Comment:
@dvoytenko I had done that, but when the assertions are in buildCallback, I get validator errors for the tests:

```
AssertionError: expected promise to be rejected with an error matching /Either the data-media-id
or the data-playlist-id attributes must be/ but it was fulfilled with <amp-jwplayer class="-amp-
element -amp-notbuilt amp-notbuilt -amp-layout-responsive -amp-layout-size-defined -amp-
error" data-player-id="sDZEo0ea" width="320" height="180" layout="responsive" id="AMP_1">
<i-amp-sizer style="display: block; padding-top: 56.25%;"></i-amp-sizer><iframe 
frameborder="0" allowfullscreen="true" src="https://content.jwplatform.com/players/undefined-
undefined.html" class="-amp-fill-content" width="320" height="180"></iframe></amp-jwplayer>
```

Should I add an additional assertion in `layoutCallback`?  Or is there an issue in the tests?